web/modules/custom/usergreeting/src/GreetingTime.php

```diff
@@ -61,34 +72,33 @@ public function __construct(TimeInterface $timeOutput, AccountInterface $account
    */
   public function greetingMessage(): array {
     $time_output = (int) date('G', $this->timeOutput->getRequestTime());
+    $config_morning = $this->configFactory->get('usergreeting.settings')->get('morning_start');
+    $config_afternoon = $this->configFactory->get('usergreeting.settings')->get('afernoon_start');
+    $config_evening = $this->configFactory->get('usergreeting.settings')->get('evening_start');
 
-    if ($time_output >= 5 && $time_output < 12) {
+    if ($time_output >= $config_morning && $time_output < $config_afternoon) {
       $greeting = $this->t('Good Morning');
     }
-    elseif ($time_output >= 12 && $time_output < 18) {
+    elseif ($time_output >= $config_afternoon && $time_output < $config_evening) {
       $greeting = $this->t('Good Afternoon');
     }
     else {
       $greeting = $this->t('Good Evening');
     }
 
     return [
-      '#type' => 'html_tag',
-      '#tag' => 'p',
-      '#value' => $greeting,
       '#cache' => [
         'contexts' => [
           'timezone',
           'user',
         ],
       ],
+      '#prefix' => $greeting,
       'username' => [
```

Comment:
You don't even need this child element. Simply move the contents of this array one level out.